test/integration/hostgroup_js_test.rb

```diff
@@ -5,6 +5,31 @@ class HostgroupJSTest < IntegrationTestWithJavascript
   #   HostgroupJSTest.test_0001_submit updates taxonomy
   extend Minitest::OptionalRetry
 
+  test 'creates a hostgroup with provisioning data' do
+    env = FactoryGirl.create(:environment)
+    os = FactoryGirl.create(:ubuntu14_10, :with_associations)
+    visit new_hostgroup_path
+
+    fill_in 'hostgroup_name', :with => 'myhostgroup1'
+    select2 env.name, :from => 'hostgroup_environment_id'
+    click_link 'Operating System'
+    wait_for_ajax
+    select2 os.architectures.first.name, :from => 'hostgroup_architecture_id'
+    wait_for_ajax
+    select2 os.title, :from => 'hostgroup_operatingsystem_id'
+    wait_for_ajax
+    select2 os.media.first.name, :from => 'hostgroup_medium_id'
+    wait_for_ajax
+    select2 os.ptables.first.name, :from => 'hostgroup_ptable_id'
+    fill_in 'hostgroup_root_pass', :with => '12345678'
+    click_button('Submit')
```

Comment:
When I remove the `wait_for_ajax` after submit line, the next assert fails (`host = Hostgroup.where(:name => "myhostgroup1").first`). I also don't see any JavaScript there, could be some kind of race condition (the hidden browser in a different process).